pkg/compactor/compactor.go

```diff
@@ -144,9 +142,9 @@ func (cp *Compactor) Compact(opts *brtypes.CompactOptions) (*brtypes.Snapshot, e
 		}
 		defer client.Close()
 
-		defragCtx, defragCancel := context.WithTimeout(ctx, opts.DefragTimeout.Duration)
-		err = etcdutil.DefragmentData(defragCtx, clientMaintenance, client, ep, cp.logger)
-		defragCancel()
+		defragCtx, defragCtxCancel := context.WithCancel(ctx)
+		defer defragCtxCancel()
```

Comment:
What's the benefit/need to create a separate context here?

Reply:
removed